packages/c/caudio/xmake.lua

```diff
@@ -0,0 +1,70 @@
+package("caudio")
+
+  set_homepage("https://github.com/R4stl1n/cAudio")
+  set_description("3D Audio Engine Based on Openal")
+
+  add_urls("https://github.com/R4stl1n/cAudio/archive/refs/tags/$(version).zip",
+           "https://github.com/R4stl1n/cAudio.git")
+  add_versions("2.3.1", "10f36cd7e1583405ade9001c3782fdf04be09f0f74e56cba23fac3a2b3ed5ae5")
+
+  add_deps("openal-soft")
+  add_links("cAudio", "Vorbis", "Ogg")
+
+  on_install("windows", "linux", "macosx", function (package)
+    local dep_dir = (package:is_arch("x64", "x86_64") and "Dependencies64" or "Dependencies")
+    local configs = {
+      "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"),
+      "-DCAUDIO_STATIC=" .. (package:config("shared") and "FALSE" or "TRUE"),
+      "-DCAUDIO_DEPENDENCIES_DIR=" .. dep_dir,
+      "-DCAUDIO_BUILD_SAMPLES=FALSE"
+    }
+
+    import("package.tools.cmake").install(package, configs, {buildir = "CMake"})
```

Comment:
cmake cannot find `Al/al.h` of `openal-soft`, we can pass `packagedeps = "openal-soft"` to force add it.

```lua
import("package.tools.cmake").install(package, configs, {buildir = "CMake", packagedeps = "openal-soft"})
```